### PR TITLE
Footer modules

### PR DESCRIPTION
**Pull request by surma**

### CODE REVIEW

File: src/_includes/footer/style.css [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+      margin-left: 0;
+    }
+    &:last-of-type {
+      margin-right: 0;
+    }
+    > a {
+      display: flex;
+      flex-direction: row;
+      align-items: center;
+      > span {
```
Comment (surma) [inline src/_includes/footer/style.css]: Not particularly proud of this hack, but not sure how to get to the `.icon` with CSS modules. Open to better suggestions.

File: src/_includes/footer/assets/email.svg [hunk 2]
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24"><path d="M20 4H4a2 2 0 0 0-2 2v12c0 1.1.9 2 2 2h16a2 2 0 0 0 2-2V6a2 2 0 0 0-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"/><path d="M0 0h24v24H0z" fill="none"/></svg>
```
Comment (jakearchibald) [inline src/_includes/footer/assets/email.svg]: Super nitty: You can remove `<path d="M0 0h24v24H0z" fill="none"/>`. It's in every material icon but it doesn't seem to do anything.

File: src/_includes/footer/assets/fullgoogle.svg [hunk 3]
```diff
@@ -0,0 +1 @@
...><path fill="#EA4335" d="M351.8 142.5c0 39-30.4 67.6-67.8 67.6s-67.8-28.7-67.8-67.6c0-39.2 30.4-67.6 67.8-67.6s67.8 28.4 67.8 67.6zm-29.7 0c0-24.3-17.6-41-38-41s-38.2 16.7-38.2 41c0 24 17.6 41 38.1 41s38.1-17 38.1-41z"/><path fill="#FBBC05" d="M499.8 142.5c0 39-30.4 67.6-67.8 67.6s-67.8-28.7-67.8-67.6c0-39.2 30.4-67.6 67.8-67.6s67.8 28.4 67.8 67.6zm-29.7 0c0-24.3-17.6-41-38-41s-38.2 16.7-38.2 41c0 24 17.6 41 38.1 41s38.1-17 38.1-41z"/><path fill="#4285F4" d="M641.7 79v121.4c0 49.9-29.4 70.4-64.3 70.4-32.7 0-52.4-22-59.9-40l25.9-10.7c4.6 11 15.9 24 34 24 22.3 0 36.1-13.8 36.1-39.6v-9.8h-1a46.1 46.1 0 0 1-35.6 15.4 67 67 0 0 1-64.7-67.3c0-38.2 31-67.9 64.7-67.9 16.2 0 29 7.2 35.6 15.1h1V79h28.2zm-26.1 63.8c0-23.8-15.9-41.3-36.1-41.3-20.5 0-37.6 17.5-37.6 41.3 0 23.5 17.1 40.7 37.6 40.7 20.2 0 36-17.2 36-40.7z"/><path fill="#34A853" d="M692 8v198h-29.7V8H692z"/><path fill="#EA4335" d="M806.7 164.8l23 15.3c-7.4 11-25.3 30-56.3 30-38.4 0-67-29.7-67-67.6 0-40.2 28.9-67.6 63.7-67.6 35 0 52.2 28 57.8 43l3.1 7.7-90.4 37.4a34.4 34.4 0 0 0 32.8 20.5c15.1 0 25.6-7.5 33.3-18.7zm-71-24.3l60.5-25.1c-3.4-8.5-13.3-14.4-25.1-14.4a37 37 0 0 0-35.3 39.5z"/><path fill="none" d="M0 0h836v272H0z"/></svg>
```
Comment (jakearchibald) [inline src/_includes/footer/assets/fullgoogle.svg]: Again, `<path fill="none" d="M0 0h836v272H0z"/></svg>` can go

File: src/_includes/footer/assets/email.svg (hunk 2 above)
Comment (jakearchibald) [inline src/_includes/footer/assets/email.svg]:
```suggestion
<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24"><path d="M20 4H4a2 2 0 0 0-2 2v12c0 1.1.9 2 2 2h16a2 2 0 0 0 2-2V6a2 2 0 0 0-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"/></svg>
```

File: src/_includes/footer/assets/fullgoogle.svg (hunk 3 above)
Comment (jakearchibald) [inline src/_includes/footer/assets/fullgoogle.svg]:
```suggestion
<svg xmlns="http://www.w3.org/2000/svg" width="836" height="272"><path fill="#4285F4" d="M108 124.7V96h96c1 5 1.6 11 1.6 17.6 0 21.5-6 48.1-24.9 67-18.4 19.2-42 29.5-73.2 29.5A106.4 106.4 0 0 1 1 105C1 47.3 49.6.1 107.5.1a100 100 0 0 1 72 29l-20.3 20.2a75.3 75.3 0 0 0-127 55.8c0 42.3 33 76.3 75.3 76.3 27.4 0 43-11 53-21a59.6 59.6 0 0 0 15.6-35.7H108z"/><path fill="#EA4335" d="M351.8 142.5c0 39-30.4 67.6-67.8 67.6s-67.8-28.7-67.8-67.6c0-39.2 30.4-67.6 67.8-67.6s67.8 28.4 67.8 67.6zm-29.7 0c0-24.3-17.6-41-38-41s-38.2 16.7-38.2 41c0 24 17.6 41 38.1 41s38.1-17 38.1-41z"/><path fill="#FBBC05" d="M499.8 142.5c0 39-30.4 67.6-67.8 67.6s-67.8-28.7-67.8-67.6c0-39.2 30.4-67.6 67.8-67.6s67.8 28.4 67.8 67.6zm-29.7 0c0-24.3-17.6-41-38-41s-38.2 16.7-38.2 41c0 24 17.6 41 38.1 41s38.1-17 38.1-41z"/><path fill="#4285F4" d="M641.7 79v121.4c0 49.9-29.4 70.4-64.3 70.4-32.7 0-52.4-22-59.9-40l25.9-10.7c4.6 11 15.9 24 34 24 22.3 0 36.1-13.8 36.1-39.6v-9.8h-1a46.1 46.1 0 0 1-35.6 15.4 67 67 0 0 1-64.7-67.3c0-38.2 31-67.9 64.7-67.9 16.2 0 29 7.2 35.6 15.1h1V79h28.2zm-26.1 63.8c0-23.8-15.9-41.3-36.1-41.3-20.5 0-37.6 17.5-37.6 41.3 0 23.5 17.1 40.7 37.6 40.7 20.2 0 36-17.2 36-40.7z"/><path fill="#34A853" d="M692 8v198h-29.7V8H692z"/><path fill="#EA4335" d="M806.7 164.8l23 15.3c-7.4 11-25.3 30-56.3 30-38.4 0-67-29.7-67-67.6 0-40.2 28.9-67.6 63.7-67.6 35 0 52.2 28 57.8 43l3.1 7.7-90.4 37.4a34.4 34.4 0 0 0 32.8 20.5c15.1 0 25.6-7.5 33.3-18.7zm-71-24.3l60.5-25.1c-3.4-8.5-13.3-14.4-25.1-14.4a37 37 0 0 0-35.3 39.5z"/></svg>
```

File: src/_includes/footer/assets/g.svg [hunk 4]
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="192" height="192"><path fill="#4285F4" d="M188 98.2c0-7-.7-13.7-1.8-20.2H96v37h52a44.6 44.6 0 0 1-19.3 29.8v24h30.9c18-16.7 28.4-41.4 28.4-70.6z"/><path fill="#34A853" d="M128.7 144.8A57.7 57.7 0 0 1 96 154c-25 0-46.3-16.9-53.9-39.6H10.2V139A96 96 0 0 0 96 192c26 0 47.7-8.5 63.6-23.2l-31-24z"/><path fill="#FBBC05" d="M39 96c0-6.4 1.2-12.6 3.1-18.4V53H10.2a95.6 95.6 0 0 0 0 86.2l31.9-24.7c-2-5.8-3-12-3-18.4z"/><path fill="#EA4335" d="M96 38c14.2 0 26.8 4.9 36.8 14.4L160.2 25A92 92 0 0 0 96 0a96 96 0 0 0-85.8 52.9l31.9 24.7A57.4 57.4 0 0 1 96 38z"/><path fill="none" d="M0 0h192v192H0z"/></svg>
```
Comment (jakearchibald) [inline src/_includes/footer/assets/g.svg]:
```suggestion
<svg xmlns="http://www.w3.org/2000/svg" width="192" height="192"><path fill="#4285F4" d="M188 98.2c0-7-.7-13.7-1.8-20.2H96v37h52a44.6 44.6 0 0 1-19.3 29.8v24h30.9c18-16.7 28.4-41.4 28.4-70.6z"/><path fill="#34A853" d="M128.7 144.8A57.7 57.7 0 0 1 96 154c-25 0-46.3-16.9-53.9-39.6H10.2V139A96 96 0 0 0 96 192c26 0 47.7-8.5 63.6-23.2l-31-24z"/><path fill="#FBBC05" d="M39 96c0-6.4 1.2-12.6 3.1-18.4V53H10.2a95.6 95.6 0 0 0 0 86.2l31.9-24.7c-2-5.8-3-12-3-18.4z"/><path fill="#EA4335" d="M96 38c14.2 0 26.8 4.9 36.8 14.4L160.2 25A92 92 0 0 0 96 0a96 96 0 0 0-85.8 52.9l31.9 24.7A57.4 57.4 0 0 1 96 38z"/></svg>
```

File: src/_includes/footer/style.css (hunk 1 above)
Comment (jakearchibald) [reply src/_includes/footer/style.css]: Usually you'd give everything a class, but this keeps the HTML pretty clean, so I'm not bothered.

File: src/_includes/footer/style.css [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+.icon-mask {
+  background: var(--midgrey);
+  mask-image: var(--icon);
+  -webkit-mask-image: var(--icon);
+  mask-size: contain;
+  -webkit-mask-size: contain;
+  mask-repeat: no-repeat;
+  -webkit-mask-repeat: no-repeat;
+  mask-position: center;
+  -webkit-mask-position: center;
```
Comment (jakearchibald) [inline src/_includes/footer/style.css]: It's usually better to put the prefixed version first so the standard version takes priority.

File: src/_includes/footer/assets/email.svg (hunk 2 above)
Comment (surma) [reply src/_includes/footer/assets/email.svg]: Is that like a watermark?

File: src/_includes/footer/style.css (hunk 5 above)
Comment (surma) [reply src/_includes/footer/style.css]: good pint

Comment (surma) [reply src/_includes/footer/style.css]: point, even

File: src/_includes/footer/assets/email.svg (hunk 2 above)
Comment (jakearchibald) [reply src/_includes/footer/assets/email.svg]: I dunno really. It might be there to define the bounds of the icon.

**Pull request merged**